src/main/java/com/Elessar/app/server/LogOnHandler.java

```diff
@@ -52,26 +56,36 @@ public void handle(HttpExchange he) throws IOException {
             final LogonRequest logonRequest = LogonRequest.parseFrom(is);
             final LogonResponse.Builder logonResponse = LogonResponse.newBuilder();
             final String userName = logonRequest.getName();
+            final String password = logonRequest.getPassword();
             final String clientURL = new StringBuilder().append("http://")
-                                                               .append(he.getRemoteAddress().getAddress().getHostAddress())
-                                                               .append(":")
-                                                               .append(logonRequest.getPort()).toString();
+                                                        .append(he.getRemoteAddress().getAddress().getHostAddress())
+                                                        .append(":")
+                                                        .append(logonRequest.getPort()).toString();
 
-            final User prevUser = db.update(new User(userName, logonRequest.getPassword(), null, null, clientURL, true));
+            final User prevUser = users.getUnchecked(userName);
+            final User currUser = new User(userName, password, null, null, clientURL, true);
 
-            if (prevUser == null) {
+            if (prevUser.getName() == null || !password.equals(prevUser.getPassword())) {
```

Comment:
since this is an invalid request, I recommend invalidate(userName) here to remove the dummy entry